cluster-api/src/main/java/io/aiven/klaw/clusterapi/services/ApacheKafkaAclService.java

```diff
@@ -0,0 +1,490 @@
+            return "Acl already exists. success";
+          }
+        } else {
+          processOtherRequests(clusterAclRequest, client, patternType, host, principal);
+        }
+        // Update transactional id acls
+        updateTransactionalIdAclsForProducer(
+            clusterAclRequest.getTransactionalId(),
+            client,
+            patternType,
+            host,
+            principal,
+            clusterAclRequest.getRequestOperationType().value);
+      }
+
+    } catch (Exception e) {
+      e.printStackTrace();
+      return ApiResultStatus.FAILURE.value;
+    }
+
+    return ApiResultStatus.SUCCESS.value;
+  }
+
+  private void processOtherRequests(
+      ClusterAclRequest clusterAclRequest,
+      AdminClient client,
+      PatternType patternType,
+      String host,
+      String principal)
+      throws InterruptedException, ExecutionException, TimeoutException {
+    List<AclBindingFilter> aclListArray = new ArrayList<>();
+
+    ResourcePatternFilter resourcePattern =
+        new ResourcePatternFilter(
+            ResourceType.TOPIC, clusterAclRequest.getTopicName(), patternType);
+    AccessControlEntryFilter aclEntry =
+        new AccessControlEntryFilter(principal, host, AclOperation.WRITE, AclPermissionType.ALLOW);
+    AclBindingFilter aclBinding1 = new AclBindingFilter(resourcePattern, aclEntry);
+    aclListArray.add(aclBinding1);
+
+    aclEntry =
+        new AccessControlEntryFilter(
+            principal, host, AclOperation.DESCRIBE, AclPermissionType.ALLOW);
+    AclBindingFilter aclBinding2 = new AclBindingFilter(resourcePattern, aclEntry);
+    aclListArray.add(aclBinding2);
+
+    client.deleteAcls(aclListArray).all().get(TIME_OUT_SECS_FOR_ACLS, TimeUnit.SECONDS);
+  }
+
+  private boolean updateTopicProducerWriteAcls(
+      String topicName, AdminClient client, PatternType patternType, String host, String principal)
+      throws InterruptedException, ExecutionException, TimeoutException {
+    List<AclBinding> aclListArray = new ArrayList<>();
+
+    ResourcePattern resourcePattern =
+        new ResourcePattern(ResourceType.TOPIC, topicName, patternType);
+    AccessControlEntry aclEntry =
+        new AccessControlEntry(principal, host, AclOperation.WRITE, AclPermissionType.ALLOW);
+    AclBinding aclBinding1 = new AclBinding(resourcePattern, aclEntry);
+    aclListArray.add(aclBinding1);
+
+    boolean acl1Exists = aclExists(client, aclBinding1.toFilter());
+
+    aclEntry =
+        new AccessControlEntry(principal, host, AclOperation.DESCRIBE, AclPermissionType.ALLOW);
+    AclBinding aclBinding2 = new AclBinding(resourcePattern, aclEntry);
+    aclListArray.add(aclBinding2);
+
+    boolean acl2Exists = aclExists(client, aclBinding2.toFilter());
+
+    if (acl1Exists && acl2Exists) {
+      return true;
+    } else {
+      client.createAcls(aclListArray).all().get(TIME_OUT_SECS_FOR_ACLS, TimeUnit.SECONDS);
+      return false;
+    }
```

Comment:
```suggestion

    if (acl1Exists && acl2Exists) return true;
    client.createAcls(aclListArray).all().get(TIME_OUT_SECS_FOR_ACLS, TimeUnit.SECONDS);
    return false;
```
the same however just a bit shorter

Reply:
I would still like to keep brackets for one line of if block. but rest looks good from your suggestion